### ISSUE TITLE
False positive unused import when referencing extension function and import is only used in KDoc

### ISSUE THREAD

**Issue by matejdro** [pinterest/ktlint#2330]
## Steps to Reproduce
Run ktlint on the following file (with coroutines on the classpath):

```kotlin
import kotlinx.coroutines.flow.Flow
import kotlinx.coroutines.flow.onStart

/**
 * [Flow.onStart]
 */
class Test
```

## Expected Behavior

Ktlint should pass

## Observed Behavior

Ktlint fails with `Unused import`

## Your Environment

* Version of ktlint used: `0.50.0` (I realise it's not the latest version, but detekt does not include later version and I don't see anything related in the changelog of the new versions)
* Name and version (or code for custom task) of integration used (Gradle plugin, Maven plugin, command line, custom Gradle task): `detekt 1.23.1`
* Version of Gradle used (if applicable): `8.4`
* Operating System and version: `Linux 6.5.5-arch1-1`

This appears to be a regression of the #1282 (I decided to create a new issue, because that one is very old)

**Comment by paul-dingemans** [comment]
Ktlint has to be conservative with removing imports. Sometimes there is just not enough information to decide whether imports can or can not be removed safely. Removing imports too eagerly results in code which can not be compiled. 

In example above, it is clear that no compilation problem will occur when imports are removed. But after removal of the imports the links in the KDocs will no longer be resolved. See screenshots below.

With both imports, the links to `Flow` and `onStart` can be clicked in IntelliJ IDEA:
![Screenshot 2023-10-23 at 20 47 09](https://github.com/pinterest/ktlint/assets/5195292/b4a3f7b3-c48c-4140-a112-4467e1b9eafe)

Removal of import `onStart` makes the `onStart` link not clickable:
![Screenshot 2023-10-23 at 20 47 36](https://github.com/pinterest/ktlint/assets/5195292/bb40c151-0eff-457f-aa9b-f5bf2eff8edd)

Removal of import `Flow` makes both links not clickable:
![Screenshot 2023-10-23 at 20 47 51](https://github.com/pinterest/ktlint/assets/5195292/91d868bd-dd7d-4585-9fb6-ebf513287dca)

So both imports in this example are needed for the KDoc (only).
